nodes/config/ui_base.html

```diff
@@ -155,7 +155,32 @@
 
 
 <script type="text/javascript">
+// #region typedefs for intellisense and better code completion/DX
+
+/**
+ * @typedef {Object} DashboardItem - A widget/group/page item that contains all of `WidgetInfo` properties
+ * @property {String} itemType - The type of item (e.g. 'widget', 'group', 'page')
+ * @property {String} id - The unique id of the item
+ * @property {String} name - The name of the item
+ * @property {String} type - The type of the item (e.g. 'ui-button', 'ui-template', 'ui-group', 'ui-page')
+ * @property {Number} order
+ * @property {String} label
+ * @property {String} icon
+ * @property {String} color
+ * @property {String} [group] - The group id that this widget belongs to
+ * @property {String} [page] - The page id that this group belongs to
+ * @property {String} [theme] - The theme id that this page belongs to
+ * @property {Boolean} [isSubflowInstance] - Whether or not this item is a subflow instance
+ * @property {Object} node - The actual node that this item represents
+ * @global
+ */
+
+/** @typedef {Object<string, Array<DashboardItem>>} DashboardItemLookup */
+// #endregion
+
 (function () {
+    const url = new URL(window.location.href)
+    window._db2debug = url.searchParams.has('debug') || url.searchParams.has('debug-ui')
```

Comment:
is this a standard NR model/practice?

Reply:
also, why `window.`, when later, it's referenced as `RED.`?

Reply:
Yep, that's an artifact I need to clean up

Reply:
resolved in [fef3ec1](https://github.com/FlowFuse/node-red-dashboard/pull/634/commits/fef3ec164c8042ce9e15e24550d7eb4faf84f0fb)